src/Raven.Server/Documents/DocumentDatabase.cs

```diff
@@ -488,12 +489,21 @@ public async Task<List<ClusterTransactionCommand.SingleClusterDatabaseCommand>>
             }
 
             var mergedCommands = new BatchHandler.ClusterTransactionMergedCommand(this, batch);
+
+            ForTestingPurposes?.BeforeExecutingClusterTransactions?.Invoke();
+
             try
             {
                 //If we get a database shutdown while we process a cluster tx command this
                 //will cause us to stop running and disposing the context while its memory is still been used by the merger execution
                 await TxMerger.Enqueue(mergedCommands);
             }
+            catch (ObjectDisposedException) when (_databaseShutdown.IsCancellationRequested)
+            {
+                HandleDatabaseShutdown(batch);
+
+                return batch;
+            }
```

Comment:
when are we getting the `ObjectDisposedException`? if now you are waiting for the task to be completed this should be thrown, right?

Reply:
txmerger is disposed first, so I assumed during review that from there

Reply:
@karmeli87 because we dispose the txmerger first.

Reply:
okay, but I think it would be better to have it the other way around...
remove this `catch` in the other ones do `catch (Exception e) when (_databaseShutdown.IsCancellationRequested == false)`
so this exception will be propagated all the way up and handled the same for all cases in one place.
Because right now if you executed one by one you will throw it N (number of commands in the batch) times

thought?

Reply:
We'll need to wait for all the transactions to be executed.

Reply:
@karmeli87 done